### PR TITLE
{bio}[foss/2018a] OpenMM v7.2.2 + OpenMM-PLUMED v20180317

### PR DESCRIPTION
**Pull request by wpoely86**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
nic157 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/38234b3be36bf0b0ba4b2388d33ff928 for a full test report.

File: easybuild/easyconfigs/o/OpenMM/OpenMM-7.2.2-fix-nvcc.patch [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
++++ openmm-7.2.2/platforms/cuda/src/CudaContext.cpp	2018-09-11 14:40:09.281703550 +0200
+@@ -199,7 +199,10 @@
+             else
+                 flags += CU_CTX_SCHED_SPIN;
+ 
+-            if (cuCtxCreate(&context, flags, device) == CUDA_SUCCESS) {
++            CUresult res = cuCtxCreate(&context, flags, device);
++            CHECK_RESULT(res);
++
++            if (res == CUDA_SUCCESS) {
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenMM/OpenMM-7.2.2-fix-nvcc.patch]: @wpoely86 This change isn't explained in the patch description?

File: easybuild/easyconfigs/o/OpenMM/OpenMM-7.2.2-foss-2018a-Python-3.6.4-CUDA-9.2.148.1.eb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+name = "OpenMM"
+version = "7.2.2"
+cuda_ver = '9.2.148.1'
+versionsuffix = '-Python-%%(pyver)s-CUDA-%s' % cuda_ver
+
+homepage = 'https://simtk.org/home/openmm'
+description = """OpenMM is a toolkit for molecular simulation."""
+
+toolchain = {'name': 'foss', 'version': '2018a'}
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenMM/OpenMM-7.2.2-foss-2018a-Python-3.6.4-CUDA-9.2.148.1.eb]: @wpoely86 Why not use `fosscuda`?

File: easybuild/easyconfigs/o/OpenMM/OpenMM-7.2.2-foss-2018a-Python-3.6.4-CUDA-9.2.148.1.eb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    ('Python', '3.6.4'),
+    ('CUDA', cuda_ver, '', True),
+    ('SWIG', '3.0.12', '-Python-%(pyver)s'),
+]
+
+# the tests take 2h
+runtest = """test -e ARGS="-E \'(TestReferenceVirtualSites|TestCudaVirtualSitesSingle|TestCudaVirtualSitesMixed|\
+TestCudaVirtualSitesDouble|TestOpenCLMonteCarloAnisotropicBarostatDouble|TestOpenCLVirtualSitesSingle|\
+TestOpenCLVirtualSitesMixed|TestOpenCLVirtualSitesDouble\
+|TestReferenceAmoebaGeneralizedKirkwoodForce|TestSystem)\'" """
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenMM/OpenMM-7.2.2-foss-2018a-Python-3.6.4-CUDA-9.2.148.1.eb]: My eyes!

The comment doesn't really explain what's going on here... Are we just running a part of the tests?

How was this determined?

**Comment by boegel** [comment]
Hmm, I'm a bit confused why the dep variants check is failing here, since we have an exception for easyconfigs that are installed for a particular version of CUDA...

@wpoely86 Can you check and see if you spot anything wrong with it? Look for `CUDA` in `test/easyconfigs/easyconfigs.py`.

**Comment by boegel** [comment]
I made a quick attempt to revive this.

@wpoely86 Do you think it's worth the effort to try and get this still merged?

**Comment by akesandgren** [comment]
I'd say no since we have 7.4.2 already.

**Comment by boegel** [comment]
@akesandgren OpenMM-PLUMED is new though.

**Comment by akesandgren** [comment]
Aha...

**Comment by wpoely86** [comment]
I have no clue to be honest. By a quick look of it, it doesn't seem hard to reproduce on a newer version?

**Comment by boegel** [comment]
I'm closing this as is, if still useful a new PR can be made for OpenMM-PLUMED.

@wpoely86 My apologies for not following up on this more closely...

**Pull request closed without merging**